dask/cli.py

```diff
@@ -91,25 +121,34 @@ def config_set(key=None, value=None):
         exit(1)
     else:
         value = dask.config.interpret_value(value)
-        _, path = save_config({key: value})
+        _, path = save_config(key, value, file)
         click.echo(f"Updated [{key}] to [{value}], config saved to {path}")
 
 
-def save_config(new_config: dict) -> tuple[dict, pathlib.Path]:
+def save_config(
+    key: str,
+    value: Any,
+    config_file: pathlib.Path | None = None,
+) -> tuple[dict, pathlib.Path]:
     """
     Save new config values to dask config file,
     return new config and path to file.
     """
-    config_file = pathlib.Path(dask.config.PATH).joinpath("dask.yaml")
+    # If explicit config file wasn't supplied, try finding the first existing
+    # config file w/ this key, falling back to the default config path
+    if config_file is None:
+        config_file = next(
+            dask.config.paths_containing_key(key),
+            pathlib.Path(dask.config.PATH).joinpath("dask.yaml"),
```

Comment:
```suggestion
            pathlib.Path(dask.config.PATH) / "dask.yaml",
```
I think this is the same and a smidge easier on the eyes?